### PR TITLE
ceph: add label for CSI deployments/daemonsets/statefulsets

### LINKED ISSUE #4120: Ceph: CSI deployments/daemonsets should have "rook-version" label
**Is this a bug report or feature request?**
* Bug Report

**Current behavior:**
```
$> kubectl -n rook-ceph get deployments -o jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \trook-version="}{.metadata.labels.rook-version}{"  \tceph-version="}{.metadata.labels.ceph-version}{"\n"}{end}'
csi-cephfsplugin-provisioner  	req/upd/avl: 2/2/2  	rook-version=  	ceph-version=
csi-rbdplugin-provisioner  	req/upd/avl: 2/2/2  	rook-version=  	ceph-version=
rook-ceph-mgr-a  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-mon-a  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-mon-b  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-mon-c  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-operator  	req/upd/avl: 1/1/1  	rook-version=  	ceph-version=
rook-ceph-osd-0  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-osd-1  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-osd-2  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-osd-3  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-osd-4  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-osd-5  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-tools  	req/upd/avl: 1/1/1  	rook-version=  	ceph-version=

$> kubectl -n rook-ceph get daemonsets -o jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \trook-version="}{.metadata.labels.rook-version}{"  \tceph-version="}{.metadata.labels.ceph-version}{"\n"}{end}'
csi-cephfsplugin  	req/upd/avl: //  	rook-version=  	ceph-version=
csi-rbdplugin  	req/upd/avl: //  	rook-version=  	ceph-version=
rook-ceph-agent  	req/upd/avl: //  	rook-version=  	ceph-version=
rook-discover  	req/upd/avl: //  	rook-version=  	ceph-version=
```

**Expected behavior:**
The CSI daemonsets and deployments should probably have the `rook-version` label added to them just as the Ceph daemons.

```
$> kubectl -n rook-ceph get deployments -o jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \trook-version="}{.metadata.labels.rook-version}{"  \tceph-version="}{.metadata.labels.ceph-version}{"\n"}{end}'
csi-cephfsplugin-provisioner  	req/upd/avl: 2/2/2  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=
csi-rbdplugin-provisioner  	req/upd/avl: 2/2/2  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=
rook-ceph-mgr-a  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-mon-a  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-mon-b  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-mon-c  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-operator  	req/upd/avl: 1/1/1  	rook-version=  	ceph-version=
rook-ceph-osd-0  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-osd-1  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-osd-2  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-osd-3  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-osd-4  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-osd-5  	req/upd/avl: 1/1/1  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=14.2.4
rook-ceph-tools  	req/upd/avl: 1/1/1  	rook-version=  	ceph-version=

$> kubectl -n rook-ceph get daemonsets -o jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \trook-version="}{.metadata.labels.rook-version}{"  \tceph-version="}{.metadata.labels.ceph-version}{"\n"}{end}'
csi-cephfsplugin  	req/upd/avl: //  	rook-version= v1.1.0-beta.0.262.gfdd92ec 	ceph-version=
csi-rbdplugin  	req/upd/avl: //  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=
rook-ceph-agent  	req/upd/avl: //  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=
rook-discover  	req/upd/avl: //  	rook-version=v1.1.0-beta.0.262.gfdd92ec  	ceph-version=
```

@travisn @Madhu-1 does this make sense to you?

**How to reproduce it (minimal and precise):**
- Install Rook with CSI enabled

### PR DESCRIPTION
**Pull request by Yisaer**
Add rook version label to CSI deployments/daemonsets/statefulsets just as the Ceph daemons.

Signed-off-by: Song Gao <2695690803@qq.com>

Description of your changes:
Add rook-version label for CSI deployments/daemonsets/statefulsets.

Which issue is resolved by this Pull Request:
Resolves #4120

**Checklist:**

- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [x] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.
[test ceph]

### CODE REVIEW

File: pkg/operator/ceph/agent/agent.go [hunk 1]
```diff
@@ -275,6 +275,7 @@ func (a *Agent) createAgentDaemonSet(namespace, agentImage, serviceAccount strin
 	} else {
 		logger.Infof("rook-ceph-agent daemonset started")
 	}
+	k8sutil.AddRookVersionLabelToDaemonSet(ds)
```
Comment (Madhu-1) [inline pkg/operator/ceph/agent/agent.go]: Why this is needed? If needed this should be a separate commit

Comment (Yisaer) [reply pkg/operator/ceph/agent/agent.go]: > Why this is needed? If needed this should be a separate commit

Resolved.

**Comment by Madhu-1** [comment]
Commit message looks same to me on both commits.

**Comment by Yisaer** [comment]
> Commit message looks same to me on both commits.

squash into one commit now.
